File: AppClasseCarro02/app/src/main/java/foo/maddo/appclassecarro02/Carro.java
package foo.maddo.appclassecarro02;

import android.util.Log;

public class Carro {

    private String fabricante;
    private String modelo;
    private String ano;
    private String cor;

    public Carro() {

        Log.i("Carro", "Carro criado");

    }

    public String getFabricante() {
        return fabricante;
    }

    public void setFabricante(String fabricante) {
        this.fabricante = fabricante;
    }

    public String getModelo() {
        return modelo;
    }

    public void setModelo(String modelo) {
        this.modelo = modelo;
    }

    public String getAno() {
        return ano;
    }

    public void setAno(String ano) {
        this.ano = ano;
    }

    public String getCor() {
        return cor;
    }

    public void setCor(String cor) {
        this.cor = cor;
    }
}

File: Upgrade2025/projetos_kotlin/AppClasseCarro02/app/src/main/java/foo/maddo/kotlin/appclassecarro02/Carro.kt
package foo.maddo.kotlin.appclassecarro02

import android.util.Log

data class Carro(
    var fabricante: String = "",
    var modelo: String = "",
    var ano: String = "",
    var cor: String = ""
) {
    init {
        Log.i("Carro", "Carro criado: $this")
    }
}
